File: wfm-develop mobile 07.05.2026/wfm/mobile/android/app/src/main/java/com/beyondviolet/wfm/core/analytics/NoOpAnalyticsService.kt
package com.beyondviolet.wfm.core.analytics

/**
 * Заглушка аналитики — используется до подключения Firebase SDK.
 * Не отправляет никаких данных.
 */
class NoOpAnalyticsService : AnalyticsService {
    override fun track(event: AnalyticsEvent) = Unit
    override fun setUser(userId: Int, role: String) = Unit
    override fun resetUser() = Unit
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMApp/Core/Analytics/NoOpAnalyticsService.swift
import Foundation

/// Заглушка аналитики — используется до подключения Firebase SDK.
/// Переключить на FirebaseAnalyticsService в DependencyContainer после добавления Firebase через SPM.
final class NoOpAnalyticsService: AnalyticsService {
    func track(_ event: AnalyticsEvent) {}
    func setUser(userId: Int, role: String) {}
    func resetUser() {}
}
